src/topology/sheaves/structure_sheaf.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Scott Morrison
+-/
+import algebraic_geometry.prime_spectrum
+import ring_theory.localization
+import algebra.category.CommRing
+import topology.sheaves.local_predicate
+import topology.sheaves.forget
+import ring_theory.subring
+
+/-!
+# The structure sheaf on `prime_spectrum R`.
+
+-/
+
+universe u
+
+noncomputable theory
+
+variables (R : Type u) [comm_ring R]
+
+open Top
+open topological_space
+open category_theory
+open opposite
+
+/--
+The type family over `prime_spectrum R` consisting of the localization over each point.
+-/
+@[reducible]
+def localizations := λ (P : Top.of (prime_spectrum R)), localization.at_prime P.as_ideal
+
+variables {R}
+
+/--
+The predicate saying that a dependent function on an open `U` is realized on `U` as a fixed fraction
+`r / s` in each of the stalks (which are localizations at various prime ideals).
+-/
+def is_fraction {U : opens (Top.of (prime_spectrum R))} (f : Π x : U, localizations R x) : Prop :=
+∃ (r s : R), ∀ x : U,
+  ¬ (s ∈ x.1.as_ideal) ∧ f x * (localization.of _).to_map s = (localization.of _).to_map r
+
+variables (R)
+
+/--
+The predicate `is_fraction` is "prelocal",
+in the sense that if it holds on `U` it holds on any open subset `V` of `U`.
+-/
+def is_fraction_prelocal : prelocal_predicate (localizations R) :=
+{ pred := λ U f, is_fraction f,
+  res := λ V U i f h,
+  begin
+    rcases h with ⟨r, s, w⟩,
+    refine ⟨r, s, λ x, w (i x)⟩,
+  end }
+
+variables {R}
+
+/--
+We will define the structure sheaf as
```

Comment:
Should some version/summary of this docstring appear in the file-level module docstring?

Reply:
Already done!